news/3931.bugfix.rst

```diff
@@ -0,0 +1 @@
+Prefer credentials from the URL over the previously-obtained credentials from URLs of the same domain, so it is possible to use different credentials on the same index server for different `--extra-index-url` options.
```

Comment:
```suggestion
Prefer credentials from the URL over the previously-obtained credentials from URLs of the same domain, so it is possible to use different credentials on the same index server for different ``--extra-index-url`` options.
```

Reply:
This is also missing a trailing newline (see failed linter check for details).

Reply:
Sure. It should be better now.